File: Cpp/CXR_classify/labelstage.cpp
#include "labelstage.h"

LabelStage::LabelStage(ConfigHandler * configHandler, DatabaseHandler * dbHandler) : Stage()
{
    if (configHandler->getDatasetType() == "subset") {
        labeler = new ManualLabeler(configHandler, dbHandler);
    } else if (configHandler->getDatasetType() == "full_set") {
        labeler = new LabelImporter(configHandler, dbHandler);
    }
}

void LabelStage::label()
{
    threadpool->start(labeler);
}

File: Go/CxrClassify/labelStage/labelStage.go
package labelStage

import (
	"CxrClassify/configHandler"
	"CxrClassify/databaseHandler"
	"CxrClassify/stage"
)

type LabelStage struct {
	stage.Stage

	ManualLabeler *ManualLabeler

	_ func() `constructor:"init"`
}

func (l *LabelStage) init() {

}

func (l *LabelStage) Setup(configHandler *configHandler.ConfigHandler, databaseHandler *databaseHandler.DatabaseHandler) {
	l.ManualLabeler = NewManualLabeler(nil)
	l.ManualLabeler.Setup(configHandler, databaseHandler)
}

func (l *LabelStage) Label() {
	l.ManualLabeler.Run()
}
